lib/OpenQA/Utils.pm

```diff
@@ -223,6 +224,25 @@ sub needle_info {
     $needle->{name}      = $name;
     $needle->{distri}    = $distri;
     $needle->{version}   = $version;
+
+    # After change the needle's properties form to hash, support backward compatibility
+    return $needle unless $needle->{properties};
+
+    # Transform string-workaround-properties into HASH-workaround-properties
+    my @properties;
+    foreach my $property (@{$needle->{properties}}) {
+        if (ref($property) eq "HASH") {
+            push(@properties, $property);
+            next;
+        }
+
+        my $desc = find_bug_number($name);
+        $desc = "" unless $desc;
+
+        push(@properties, {name => $property, value => $desc});
+    }
+    $needle->{properties} = \@properties;
```

Comment:
that looks like it could be improved with a call to `map`. Given that your unit tests should cover that it should not be too hard to try out :)

Reply:
Yea, should be possible to do with a `map` one-liner.